CI.hs

```diff
@@ -96,11 +111,14 @@ bintrayUpload version = do
             putStrLn $ "# Completed in " ++ showDuration t ++ ": " ++ c ++ "\n"
             hFlush stdout
 
-buildDists :: GhcFlavor -> IO String
-buildDists ghcFlavor = do
+buildDists :: GhcFlavor -> Maybe String -> IO String
+buildDists ghcFlavor resolver = do
+    -- One of "" (use 'stack.yaml') or, "--resolver=xxx".
+    let resolverFlag=stackResolverOpt resolver
+
     -- Get packages missing on Windows needed by hadrian.
     when isWindows $
-        cmd "stack exec -- pacman -S autoconf automake-wrapper make patch python tar --noconfirm"
+        cmd $ "stack exec "  ++ resolverFlag ++ " -- pacman -S autoconf automake-wrapper make patch python tar --noconfirm"
```

Comment:
Maybe it would make sense to define something like `let stackCmd cmd = cmd $ "stack " ++ resolverFlag ++ " " ++ cmd` at the beginning of this function and then use that everywhere. That makes it a bit easier to set default flags that should be passed to all Stack invocations.

Reply:
Indeed. I'll look into cleaning this up today.